core/src/main/scala/chisel3/Data.scala

```diff
@@ -832,6 +832,9 @@ object Data {
       }
   }
 
+  /** Useful for reflect.DataMirror collect iterating functions */
+  val DataMatchingZipOfChildren = implicitDataMatchingZipOfChildren
```

Comment:
From what I remember, @jackkoenig didn't like the public implicit val on the object. But, in order to use the collect functions on DataMirror, you need a manifestation of the object. I don't think it was possible to use those functions prior to this PR outside of chisel3, but they are super useful.

Reply:
Yeah it definitely shouldn't be publicly implicit. Can you give a more useful ScalaDoc and a type annotation since this is a new public API?

Reply:
After chatting with Adam, we should just make the main implicit public as it is the canonical implementation of a typeclass for `Data`, but it would still benefit from a type annotation.